src/journeys_utils.js

```diff
@@ -438,14 +439,94 @@ journeys_utils.animateBannerEntrance = function(banner) {
 			banner.style.top = '0';
 		}
 		else if (journeys_utils.position === 'bottom') {
-			banner.style.bottom = '0';
+			if(!journeys_utils._isSafeAreaRequired(journeys_utils.journeyLinkData)) {
+				banner.style.bottom = '0';
+			} else {
+				journeys_utils._dynamicallyRepositionBanner(banner);
+			}
 		}
         journeys_utils.branch._publishEvent('didShowJourney', journeys_utils.journeyLinkData);
 		journeys_utils.isJourneyDisplayed = true;
 	}
 	setTimeout(onAnimationEnd, journeys_utils.animationDelay);
 }
 
+journeys_utils._isSafeAreaRequired = function(journeyLinkData) {
+	if (journeyLinkData['safeAreaRequired'] && journeys_utils._detectSafeAreaDevices()) {
+		return true;
+	}
+	return false;
+}
+
+journeys_utils._resizeListener = function () {
+	if (journeys_utils.isSafeAreaEnabled) {
+		journeys_utils._resetJourneysBannerPosition(false);
+	}
+}
+
+journeys_utils._scrollListener = function () {
+	if (journeys_utils.isSafeAreaEnabled) {
+		if (window.pageYOffset > window.innerHeight) {
+			journeys_utils._resetJourneysBannerPosition(true);
+		} else {
+			journeys_utils._resetJourneysBannerPosition(false);
+		}
+	}
+}
+
+journeys_utils._dynamicallyRepositionBanner = function() {
+	journeys_utils.isSafeAreaEnabled = true;
+	// disable Journey animation to avoid lag when repositioning the banner 
+	document.getElementById('branch-banner-iframe').style.transition = "all 0s"
+	// make sure on the first journey load the position is correct
+	journeys_utils._resetJourneysBannerPosition(false, true);
```

Comment:
@code-crusher on first load, why would the page already be scrolling?

Reply:
Ah ignore, it's a signal to the function to check whether the page is already scrolling.